query-algebrizer/src/clauses/pattern.rs

```diff
@@ -238,8 +238,29 @@ impl ConjoiningClauses {
                 if value_type.is_none() {
                     self.wheres.add_intersection(ColumnConstraint::HasType(col.clone(), typed_value_type));
                 }
+            },
+        }
 
+        match pattern.tx {
+            PatternNonValuePlace::Placeholder => (),
+            PatternNonValuePlace::Variable(ref v) => {
+                self.bind_column_to_var(schema, col.clone(), DatomsColumn::Tx, v.clone());
+            },
+            PatternNonValuePlace::Entid(entid) => {
+                // TODO: we want to check whether the tx-id is within range for the database's tx partition. 
+                // (That applies after ident lookup, too.)
+                // Possible solution: https://github.com/mozilla/mentat/tree/fluffyemily/tx-id-check
```

Comment:
Yeah, that's pretty messy! 😁 

The only way I think it would be less messy is for the CC to have a handle to the DB, not the schema.

Thanks for giving it a shot.

But two things cause me to disagree with myself:

- Not everything is worth checking before executing the query.
- Algebrizing with respect to a schema means that only schema changes invalidate the output. Algebrizing with respect to the partition map means that any entity allocation invalidates the output! That's probably acceptable for `q_once`, but not useful for prepared queries.